libs/pages/clusters/src/lib/ui/page-clusters-create/step-kubeconfig/step-kubeconfig.tsx

```diff
@@ -0,0 +1,111 @@
+import { type FormEventHandler, useCallback } from 'react'
+import { useDropzone } from 'react-dropzone'
+import { useController, useFormContext } from 'react-hook-form'
+import { useNavigate, useParams } from 'react-router-dom'
+import { type ClusterKubeconfigData } from '@qovery/shared/interfaces'
+import { CLUSTERS_CREATION_GENERAL_URL, CLUSTERS_CREATION_URL, CLUSTERS_URL } from '@qovery/shared/routes'
+import { Button, Dropzone, Icon, IconAwesomeEnum } from '@qovery/shared/ui'
+
+export interface StepKubeconfigProps {
+  onSubmit: FormEventHandler<HTMLFormElement>
+}
+
+export function StepKubeconfig({ onSubmit }: StepKubeconfigProps) {
+  const navigate = useNavigate()
+  const { organizationId = '' } = useParams()
+  const { control, formState } = useFormContext<ClusterKubeconfigData>()
+  const { field: fileName } = useController<ClusterKubeconfigData>({
+    name: 'file_name',
+    control,
+    rules: { required: true },
+  })
+  const { field: fileContent } = useController<ClusterKubeconfigData>({
+    name: 'file_content',
+    control,
+    rules: { required: true },
+  })
+  const { field: fileSize } = useController<ClusterKubeconfigData>({
+    name: 'file_size',
+    control,
+    rules: { required: true },
+  })
+
+  const onDrop = useCallback((acceptedFiles: File[]) => {
+    acceptedFiles.forEach((file) => {
+      const reader = new FileReader()
+
+      reader.onabort = () => console.log('file reading was aborted')
+      reader.onerror = () => console.log('file reading has failed')
+      reader.onload = () => {
+        fileSize.onChange(file.size / 1000)
+        fileName.onChange(file.name)
+        fileContent.onChange(reader.result)
```

Comment:
I think here it's better if you use an `methods.setValue` because if I go back without submit the form I lost my file and you can remove remove this method as well

``` 
 const { field: fileContent } = useController<ClusterKubeconfigData>({
    name: 'file_content',
    control,
    rules: { required: true },
  })
  const { field: fileSize } = useController<ClusterKubeconfigData>({
    name: 'file_size',
    control,
    rules: { required: true },
  })
```

Reply:
using `setValue` is not possible because of typing / form state issues.